db/common_test.go

```diff
@@ -0,0 +1,187 @@
+// Package db - common test
+// The following file contains all the common functions for testing.
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/packethost/pkg/log"
+)
+
+var (
+	ids = testID{
+		instanceID: "da013837-a30c-237a-adfc-2d9ad69d5000",
+		hardwareID: "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+		templateID: "da013837-a30c-237a-adfc-2d9ad69d5011",
+		workflowID: "b356eede-22c1-11eb-8c17-0242ac120004",
+	}
+)
+
+type hardwarePayload struct {
+	ID       string `json:"id"`
+	Metadata struct {
+		Facility struct {
+			FacilityCode string `json:"facility_code"`
+		} `json:"facility"`
+		Instance struct {
+			IPAddresses []struct {
+				Address string `json:"address"`
+			} `json:"ip_addresses"`
+		} `json:"instance"`
+		State string `json:"state"`
+	} `json:"metadata"`
+	Network struct {
+		Interfaces []struct {
+			Dhcp struct {
+				Arch string `json:"arch"`
+				IP   struct {
+					Address string `json:"address"`
+					Gateway string `json:"gateway"`
+					Netmask string `json:"netmask"`
+				} `json:"ip"`
+				Mac  string `json:"mac"`
+				Uefi bool   `json:"uefi"`
+			} `json:"dhcp"`
+			Netboot struct {
+				AllowPxe      bool `json:"allow_pxe"`
+				AllowWorkflow bool `json:"allow_workflow"`
+			} `json:"netboot"`
+		} `json:"interfaces"`
+	} `json:"network"`
+}
+
+type testArgs struct {
+	host         string
+	port         int32
+	tinkDb       *TinkDB
+	hardwareID   string
+	user         string
+	password     string
+	dbname       string
+	psqlInfo     string
+	logger       log.Logger
+	ctx          ctx
+	instanceData string
+	hardwareData string
+	// templateData - referenced in workflow.Workflow. Formatted as yaml.
+	templateData string
+	workflowData Workflow
+}
+
+type testID struct {
+	workflowID string
+	instanceID string
+	hardwareID string
+	templateID string
+}
+
+type ctx struct {
+	context context.Context
+	cancel  context.CancelFunc
+}
+
+func newCTX() ctx {
+	c, _ := context.WithCancel(context.Background())
+	return ctx{
+		context: c,
+	}
+}
```

Comment:
lets just drop this all together can use `context.Background()` in common_test.go

Reply:
dropped